components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/wsdl/util/SOAPOperationBindingUtils.java

```diff
@@ -110,8 +110,24 @@ public static String getSoapOperationMapping(byte [] wsdlContent) throws APIMana
      * @return swagger json string with the soap operation mapping
      * @throws APIManagementException if an error occurs when generating swagger
      */
-    public static String getSoapOperationMapping(String path) throws APIManagementException {
-        WSDL11SOAPOperationExtractor processor = APIMWSDLReader.getWSDLSOAPOperationExtractor(path);
+    public static String getSoapOperationMapping(String path, String url) throws APIManagementException {
+        APIMWSDLReader wsdlReader = new APIMWSDLReader(url);
+        WSDL11SOAPOperationExtractor processor = APIMWSDLReader.getWSDLSOAPOperationExtractor(path, wsdlReader);
+        WSDLInfo wsdlInfo = processor.getWsdlInfo();
+        return getGeneratedSwaggerFromWSDL(wsdlInfo);
+    }
+
+    /**
+     * Gets soap operations to rest resources mapping for a wsdl archive path
+     *
+     * @param path Path of the extracted WSDL archive
+     * @return swagger json string with the soap operation mapping
+     * @throws APIManagementException if an error occurs when generating swagger
+     */
+    public static String getSoapOperationMapping(String path, String url, byte [] wsdlContent)
+            throws APIManagementException {
+        APIMWSDLReader wsdlReader = new APIMWSDLReader(url);
+        WSDL11SOAPOperationExtractor processor = APIMWSDLReader.getWSDLSOAPOperationExtractor(path, wsdlReader);
         WSDLInfo wsdlInfo = processor.getWsdlInfo();
         return getGeneratedSwaggerFromWSDL(wsdlInfo);
     }
```

Comment:
What is the usage of this method?
byte [] wsdlContent is also never used inside. I don't see any difference with this with the method which is just above.

Reply:
Yes, I never used this method. @Chamindu36 Can u remove this method?

Reply:
Removed with b736f4e7253bd8f35a7841654afe80b3a4acba3c